test/hotspot/jtreg/compiler/loopopts/superword/TestDependencyOffsets.java

```diff
@@ -486,15418 +409,475 @@
-        applyIfAnd = {"AlignVector", "true", "MaxVectorSize", ">= 16", "MaxVectorSize", "<= 24"},
-        applyIfCPUFeatureAnd = {"avx2", "true", "avx512", "false"})
-    // CPU: avx512 -> vector_width: 64 -> elements in vector: 8
-    //   positive byte_offset 24 can lead to cyclic dependency
-    @IR(counts = {IRNode.LOAD_VECTOR_L, "> 0", IRNode.ADD_VL, "> 0", IRNode.STORE_VECTOR, "> 0"},
-        applyIfAnd = {"AlignVector", "false", "MaxVectorSize", ">= 16", "MaxVectorSize", "<= 24"},
-        applyIfCPUFeature = {"avx512", "true"})
-    //   Expect alignment.
-    @IR(counts = {IRNode.LOAD_VECTOR_L, "> 0", IRNode.ADD_VL, "> 0", IRNode.STORE_VECTOR, "> 0"},
-        applyIfAnd = {"AlignVector", "true", "MaxVectorSize", ">= 16", "MaxVectorSize", "<= 24"},
-        applyIfCPUFeature = {"avx512", "true"})
-    // CPU: asimd -> vector_width: 16 -> elements in vector: 2
-    @IR(counts = {IRNode.LOAD_VECTOR_L, "> 0", IRNode.ADD_VL, "> 0", IRNode.STORE_VECTOR, "> 0"},
-        applyIfAnd = {"AlignVector", "false", "MaxVectorSize", ">= 16"},
-        applyIfCPUFeatureAnd = {"asimd", "true", "sve", "false"})
-    //   Expect alignment.
-    @IR(counts = {IRNode.LOAD_VECTOR_L, "> 0", IRNode.ADD_VL, "> 0", IRNode.STORE_VECTOR, "> 0"},
-        applyIfAnd = {"AlignVector", "true", "MaxVectorSize", ">= 16"},
-        applyIfCPUFeatureAnd = {"asimd", "true", "sve", "false"})
-    // CPU: sve -> max vector_width: 256 -> max elements in vector: 32
-    //   positive byte_offset 24 can lead to cyclic dependency
-    @IR(counts = {IRNode.LOAD_VECTOR_L, "> 0", IRNode.ADD_VL, "> 0", IRNode.STORE_VECTOR, "> 0"},
-        applyIfAnd = {"AlignVector", "false", "MaxVectorSize", ">= 16", "MaxVectorSize", "<= 24"},
-        applyIfCPUFeature = {"sve", "true"})
-    //   Expect alignment.
-    @IR(counts = {IRNode.LOAD_VECTOR_L, "> 0", IRNode.ADD_VL, "> 0", IRNode.STORE_VECTOR, "> 0"},
-        applyIfAnd = {"AlignVector", "true", "MaxVectorSize", ">= 16", "MaxVectorSize", "<= 24"},
-        applyIfCPUFeature = {"sve", "true"})
-    public static void testLongP3(long[] data) {
-        for (int j = 0; j < RANGE - 3; j++) {
-            data[j + 3] = (long)(data[j] + (long)-11);
-        }
-    }
+        void generate(StringBuilder builder) {
+            if (!isEnabled) {
+                builder.append("    // No IR rule: disabled.\n");
+            } else {
+                builder.append(counts());
 
-    @Run(test = "testLongP3")
-    @Warmup(0)
-    public static void runLongP3() {
-        long[] data = new long[RANGE];
-        init(data);
-        testLongP3(data);
-        verify("testLongP3", data, goldLongP3);
-    }
+                // applyIf
+                if (!applyIf.isEmpty()) {
+                    builder.append("        applyIf");
+                    builder.append(applyIf.size() > 1 ? "And" : "");
+                    builder.append(" = {");
+                    builder.append(applyIf.stream().collect(Collectors.joining(", ")));
```

Comment:
Can be replaced with:
```suggestion
                    builder.append(String.join(", ", applyIf));
```